### PR TITLE
feature/#672 added doc for new submittable api

### PR DESCRIPTION
**Pull request by toan-quach**

### CODE REVIEW

File: docs/relnotes.md [hunk 1]
```diff
@@ -22,7 +22,14 @@ Not published yet.
 3.0.0
 
 - New exposed function `taipy.exists()^` checks if an entity exists or not.
-
+- New exposed class `Submittable` representing if an entity can be submitted for execution.
```
Comment (jrobinAV) [inline docs/relnotes.md]:
```suggestion
- New abstract class `Submittable^` to model entities that can be submitted for execution.
```

File: docs/relnotes.md [hunk 2]
```diff
@@ -22,7 +22,14 @@ Not published yet.
...
+  Entities that are child class of `Submittable` are `Scenario` and `Pipeline`.
```
Comment (jrobinAV) [inline docs/relnotes.md]:
```suggestion
  The children entity classes are `Scenario` and `Pipeline`.
```

File: docs/relnotes.md [hunk 3]
```diff
@@ -22,7 +22,14 @@ Not published yet.
...
+- New exposed function `Submittable.get_inputs()` retrieves input data nodes of a `Submittable` entity.
```
Comment (jrobinAV) [inline docs/relnotes.md]:
```suggestion
- New exposed function `Submittable.get_inputs()^` retrieves input data nodes of a `Submittable` entity.
```

File: docs/relnotes.md [hunk 4]
```diff
@@ -22,7 +22,14 @@ Not published yet.
...
+- New exposed function `Submittable.get_outputs()` retrieves output data nodes of a `Submittable` entity.
```
Comment (jrobinAV) [inline docs/relnotes.md]:
```suggestion
- New exposed function `Submittable.get_outputs()^` retrieves output data nodes of a `Submittable` entity.
```

File: docs/relnotes.md [hunk 5]
```diff
@@ -22,7 +22,14 @@ Not published yet.
...
+- New exposed function `Submittable.get_intermediate()` retrieves intermediate data nodes of a `Submittable` entity.
```
Comment (jrobinAV) [inline docs/relnotes.md]:
```suggestion
- New exposed function `Submittable.get_intermediate()^` retrieves intermediate data nodes of a `Submittable` entity.
```

File: docs/relnotes.md [hunk 6]
```diff
@@ -22,7 +22,14 @@ Not published yet.
...
+- New exposed function `Submittable.is_ready_to_run()` checks if an entity is ready to be run.
```
Comment (jrobinAV) [inline docs/relnotes.md]:
```suggestion
- New exposed function `Submittable.is_ready_to_run()^` checks if an entity is ready to be run.
```

File: docs/relnotes.md [hunk 7]
```diff
@@ -22,7 +22,14 @@ Not published yet.
...
+- New exposed function `Submittable.data_nodes_being_edited()` retrieves data nodes that are being editted
```
Comment (jrobinAV) [inline docs/relnotes.md]:
```suggestion
- New exposed function `Submittable.data_nodes_being_edited()^` retrieves data nodes that are being editted
```

**Review by jrobinAV — COMMENTED** [review]
I suggest to add cros ref links, but I did not check the syntax is correct.
@toan-quach can you build the doc locally to test the cross reference links ?
Thx.

**Comment by toan-quach** [comment]
@jrobinAV 
Sure I can try :D thing is that my build always fails after some changes for some reason so it's quite difficult to build the doc on my side. But I can give it a try :D

File: docs/relnotes.md [hunk 8]
```diff
@@ -21,8 +21,26 @@ Not published yet.
...
+
+       * `Submittable.get_inputs()^` retrieves input data nodes of a `Submittable` entity;
+
+       * `Submittable.get_outputs()^` retrieves output data nodes of a `Submittable` entity;
+
+       * `Submittable.get_intermediate()^` retrieves intermediate data nodes of a `Submittable` entity;
+
+       * `Submittable.is_ready_to_run()^` checks if an entity is ready to be run;
+
+       * `Submittable.data_nodes_being_edited()^` retrieves data nodes that are being editted
```
Comment (erikfTW) [inline docs/relnotes.md]:
```suggestion
       * `Submittable.data_nodes_being_edited()^` retrieves data nodes that are being edited
```

**Review by jrobinAV — APPROVED** [review]

**Pull request merged**